internal/asm/f32/ge_noasm.go

```diff
@@ -0,0 +1,43 @@
+// Copyright ©2017 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//+build !go1.8 !amd64 noasm appengine
+
+package f32
+
+// Ger performs the rank-one operation
+//  A += alpha * x * y^T
+// where A is an m×n dense matrix, x and y are vectors, and alpha is a scalar.
+func Ger(m, n uintptr, alpha float32,
+	x []float32, incX uintptr,
+	y []float32, incY uintptr,
+	a []float32, lda uintptr) {
+
+	if incX == 1 && incY == 1 {
+		x = x[:m]
+		y = y[:n]
+		for i, xv := range x {
+			AxpyUnitary(alpha*xv, y, a[uintptr(i)*lda:uintptr(i)*lda+n])
+		}
+		return
+	}
+
+	var ky, kx uintptr
+	if int(incY) > 0 {
+		ky = 0
```

Comment:
I know this is probably a copy/paste but we could clean this up a bit. It is not necessary to set `kx`, `ky` to zero, and if incY is negative, the formulas can be slightly simplified to `kx = (1 - m) * incX`.